operations/_scripts/deploy/deploy.sh

```diff
@@ -30,6 +30,14 @@ if [[ "$CREATE_DOMAIN" == "true" ]]; then
   /bin/bash $GITHUB_ACTION_PATH/operations/_scripts/generate/generate_acm_tf.sh
 fi
 
+# Ansible related generation/resourcing/configurations block
+# -------------------------------- #
+if [[ -n $ST2_ANSIBLE_VARS_FILE ]]; then
+    cp $GITHUB_WORKSPACE/$ST2_CONF_PATH $GITHUB_ACTION_PATH/operations/deployment/ansible/vars/
```

Comment:
Wouldn't this just overwrite the https://github.com/bitovi/github-actions-deploy-stackstorm/blob/main/operations/deployment/ansible/vars/st2.yaml file if user creates `st2.yaml`?

If so, that may break the deployment, for instance https://github.com/bitovi/github-actions-deploy-stackstorm/blob/main/operations/deployment/ansible/vars/st2.yaml#L42-L45 which takes ENV var from the GH Actions configuration via ENV var.

I'm still willing if we'd take the route of Ansible vars precedence, instead of just overwriting the yaml file.

default [ansible-st2 vars](https://github.com/stackStorm/ansible-st2#variables) < ours [st2.yaml](https://github.com/bitovi/github-actions-deploy-stackstorm/blob/main/operations/deployment/ansible/vars/st2.yaml) < user's `st2.yaml`

Reply:
The question is what's the best place for users provided st2.yaml vars to make it a highest var priority?

play vars_files, include_vars or something?
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence
It's something for you to explore